### PR TITLE
A4A: Implement Pressable overview page

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR implements the Pressable overview page based on the latest Purchase flow design. **Note that purchasing a plan is currently unavailable as the backend is being worked on. You may be able to add Pressable plans in the shopping cart, but you won't be able to obtain a license if you proceed to checkout.**

**Selecting a plan**
Screenshot 2024-03-26 at 2 49 21 PM
**Selecting 'More**'
Screenshot 2024-03-26 at 2 49 59 PM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/74

## Proposed Changes

* Update the Slider component to accept a null value. This is something I wasn't able to consider as a valid use case earlier.
* Implement the Pressable page based on the latest design. Since the Figma design doesn't give us full information about what each Pressable plan includes, I have to rely on the existing Pressable pricing page for this information. https://pressable.com/pricing/

## Testing Instructions

* Use the live link and go to `/marketplace/hosting/pressable`
* Confirm that the page is rendered properly based on the latest design.
* Confirm that the slider works as expected.
* Confirm that selecting a plan will add it to the Shopping cart.
* Confirm that the page is responsive.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/marketplace/pressable-overview/plan-selection/style.scss [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	display: flex;
+	flex-direction: row;
+	gap: 16px;
+
+	align-items: center;
+	justify-content: center;
+}
+
+.components-button.pressable-overview-plan-selection__filter-button {
+	border: 1.5px solid var(--studio-black);
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/pressable-overview/plan-selection/style.scss:114]: No equivalent theme properties for studio black so I am force to use it directly.

File: client/a8c-for-agencies/sections/marketplace/pressable-overview/plan-selection/details.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+				{ ! selectedPlan && (
+					<Button
+						className="pressable-overview-plan-selection__details-card-cta-button"
+						onClick={ onChatWithUs }
+						href={ PRESSABLE_CONTACT_LINK }
+						target="_blank"
+						variant="primary"
+					>
+						{ translate( 'Chat with us' ) } <Icon icon={ external } size={ 16 } />
+					</Button>
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/pressable-overview/plan-selection/details.tsx:128]: This is not 100% clear to me. I am basing the button on Pressable's existing pricing page, found at https://pressable.com/pricing/. Let me know what you think.

**Comment by simonktnga8c** [comment]
Looks good to me @jkguidaven 👍 

The only issues I can see are that there should be an "s" at the end of "WordPress Installs" and "Number of visits".

**Comment by jkguidaven** [comment]
> Looks good to me @jkguidaven 👍
> 
> The only issues I can see are that there should be an "s" at the end of "WordPress Installs" and "Number of visits".

Thanks for catching this. This has been fixed 👍

**Review by madebynoam — CHANGES_REQUESTED** [review]
Thanks for this @jkguidaven!

A small detail: On the containers we should have a 4px rounded corners:
image

On my live link, the slider only shows "More".
image

I'll try to test locally, to see if it works.

**Comment by jkguidaven** [comment]
> A small detail: On the containers we should have a 4px rounded corners:

Good catch! This has been addressed in #88889. 👍 

> On my live link, the slider only shows "More".

I apologize for not mentioning this in the instructions. You need to sign up as an A4A Agency to access the data.  

Can you try to sign up in A4A: `/signup`.
This should set up your account. Also after signing up, please confirm with @Automattic/jetpack-Avalon if your account is non-billable so you won't accidentally get charged.

**Comment by jkguidaven** [comment]
Hi everyone,

I have to split this PR into four parts to make it much easier for devs to review. See below:

#88887
#88888
#88889
#88890

In that case, I am closing this PR.

**Pull request closed without merging**